components/relevancy/src/lib.rs

```diff
@@ -95,6 +97,101 @@ impl RelevancyStore {
+    /// of success. The arm with the highest sampled probability is selected and returned.
+    ///
+    /// # Parameters
+    /// - `bandit` (String): The identifier for the bandit model used to select an arm.
+    /// - `arms` (&[String]): A list of arms (items) from which the best candidate is chosen.
+    ///
+    /// # Returns
+    /// - `ApiResult<String>`: The identifier of the selected arm with the highest sampled probability,
+    /// or an error if data retrieval fails.
+    #[handle_error(Error)]
+    pub fn bandit_select(&self, bandit: String, arms: &[String]) -> ApiResult<String> {
+        // we should cache the distribution so we don't retrieve each time
+
+        let mut best_sample = f64::MIN;
+        let mut selected_arm = String::new();
+
+        for arm in arms {
+            let (alpha, beta) = self
+                .db
+                .read(|dao| dao.retrieve_bandit_arm_beta_distribution(&bandit, arm))?;
+            // this creates a Beta distribution for an alpha & beta pair
+            let beta_dist = Beta::new(alpha as f64, beta as f64)
+                .expect("computing betas dist unexpectedly failed");
+
+            // Sample from the Beta distribution
+            let sampled_prob = beta_dist.sample(&mut rand::thread_rng());
+
+            if sampled_prob > best_sample {
+                best_sample = sampled_prob;
+                selected_arm.clone_from(arm);
+            }
+        }
+
+        return Ok(selected_arm);
+    }
+
+    /// Updates the bandit model's arm data based on user interaction (selection or non-selection).
+    ///
+    /// This method takes in a `bandit` identifier, an `arm` identifier, and a `selected` flag.
+    /// If `selected` is true, it updates the model to reflect a successful selection of the arm,
+    /// reinforcing its positive reward probability. If `selected` is false, it updates the
+    /// beta (failure) distribution of the arm, reflecting a lack of selection and reinforcing
+    /// its likelihood of a negative outcome.
+    ///
+    /// # Parameters
+    /// - `bandit` (String): The identifier for the bandit model to update.
+    /// - `arm` (String): The identifier of the specific arm within the bandit model to update.
+    /// - `selected` (bool): A flag indicating whether the arm was selected (true) or not (false).
+    ///
+    /// # Returns
+    /// - `ApiResult<()>`: An empty result if the update is successful, or an error if the transaction fails.
+    #[handle_error(Error)]
+    pub fn bandit_update(&self, bandit: String, arm: String, selected: bool) -> ApiResult<()> {
+        if selected {
+            self.db
+                .read_write(|dao| dao.update_bandit_arm_data(&bandit, &arm, selected))?;
+        } else {
+            // update beta (failure) distribution (+1) for item
+            self.db
+                .read_write(|dao| dao.update_bandit_arm_data(&bandit, &arm, selected))?;
+        }
```

Comment:
Now that `update_bandit_arm_data()` inputs `selected`, I don't think there's any need for this `if`.

Reply:
good catch!